arches_component_lab/src/arches_component_lab/cards/api.ts

```diff
@@ -40,34 +42,66 @@ export const fetchTileData = async (
     }
 };
 
-export const upsertTile = async (
+export const fetchCardXNodeXWidgetDataFromNodeGroup = async (
     graphSlug: string,
     nodegroupAlias: string,
-    data: Record<string, unknown>,
-    tileId?: string,
 ) => {
-    const args = [graphSlug, nodegroupAlias];
+    const response = await fetch(
+        arches.urls.api_card_x_node_x_widget_list_from_nodegroup(
+            graphSlug,
+            nodegroupAlias,
+        ),
+    );
+
+    try {
+        const parsed = await response.json();
+        if (response.ok) {
+            return parsed;
+        }
+        throw new Error(parsed.message);
+    } catch (error) {
+        throw new Error((error as Error).message || response.statusText);
+    }
+};
+
+export async function upsertTile(
+    graphSlug: string,
+    nodegroupAlias: string,
+    payload: Record<string, unknown>,
+    tileId?: string,
+): Promise<unknown> {
+    const urlSegments = [graphSlug, nodegroupAlias];
+    let httpMethod = "POST";
 
     if (tileId) {
-        args.push(tileId);
+        urlSegments.push(tileId);
+        httpMethod = "PUT";
```

Comment:
The reason I was using PATCH before was because it maps more closely to what the underlying save() method is doing by skipping over nodes missing from the request instead of resetting them to their default value.

Would we want to add some functionality in the future to reset nodes absent from the request to their default value? That sounds like PUT behavior if so.

Also, what to expact when you post back a resource with only some tiles present? Did you intend to delete the tiles not present in the json? If so, that sounds like PUT behavior. I don't think that's what we want in most cases, so PATCH seemed a little safer.

We can punt, but curious to pull in @apeters also for a gut check

Reply:
https://github.com/archesproject/arches-querysets/pull/70